src/database/MoneyTransactions/index.ts

```diff
@@ -0,0 +1,107 @@
+import dbConnection, { tableNames, execute } from '../connection';
+
+
+const moneyTransactionsTable = () => dbConnection()(tableNames.MONEY_TRANSACTIONS);
+
+/**
+ * Represent a money transaction after retrieval from the database
+ */
+export interface MoneyTransactionDb {
+  moneytransactionuuid: string;
+  datetime: string;
+  topersonuuid: string;
+  frompersonuuid: string;
+  amount: number;
+  currency: string;
+
+  fromfirstname: string;
+  frommiddlename: string;
+  fromlastname: string;
+
+  tofirstname: string;
+  tomiddlename: string;
+  tolastname: string;
+}
+
+/**
+ * Represent a money transaction used in an INSERT call on the database
+ */
+export interface moneyTransactionsDbInsertReq {
+  moneytransactionuuid?: string;
+  datetime: string;
+  topersonuuid: string;
+  frompersonuuid: string;
+  amount: number;
+  currency: string;
+}
+
+const builders = {
+  /** Get all money transactions of a certain type */
+  getMoneyTransactions() {
+    return moneyTransactionsTable()
+    .select(tableNames.MONEY_TRANSACTIONS + '.*',
+      'fromperson.firstname as fromfirstname',
+      'fromperson.middlename as frommiddlename',
+      'fromperson.lastname as fromlastname',
+      'toperson.firstname as tofirstname',
+      'toperson.middlename as tomiddlename',
+      'toperson.lastname as tolastname',
+      tableNames.PRODUCT_TYPES + '.*',
+  )
+    .join(tableNames.PEOPLE + ' as fromperson',
+      tableNames.MONEY_TRANSACTIONS + '.frompersonuuid',
+      'fromperson.personuuid')
+      .join(tableNames.PEOPLE + ' as toperson',
+      tableNames.MONEY_TRANSACTIONS + '.topersonuuid',
+      'toperson.personuuid')
+    .orderBy('fromlastname', 'asc')
+    .orderBy('datetime', 'asc')
+    ;
+  },
+
+  /** inserts a MoneyuctTransaction row in the database */
+  insertMoneyTransaction(transaction: moneyTransactionsDbInsertReq) {
+    return moneyTransactionsTable()
+    .returning('moneytransactionuuid')
+    .insert(transaction);
+  },
+
+  /** updates a single money field in the database */
+  updateMoneyTransactionField(moneytransactionuuid: string, field: string, value: any) {
+    return moneyTransactionsTable()
+      .update(field, value)
+      .where({ moneytransactionuuid });
+  },
+
+  /** Deletes a single money transaction row, use in case of error */
+  deleteMoneyTransaction(moneytransactionuuid: string) {
```

Comment:
oh? i thought we aren't supporting delete?

Reply:
I need it when creating a loan entry, because I need to add the money transaction before the loan entry due to foreign key constraint, if the loan entry fails than i need to delete the money row. There's no exposed API for deletion.

Reply:
ohh okay so it's only when something fails. In the other cases, we aren't deleting any transactions right? like when a loan is paid off? we will just be calculating it on the client?